internal/service/subscription.go

```diff
@@ -3517,13 +3543,103 @@ func (s *subscriptionService) addAddonToSubscription(
 	s.Logger.Infow("added addon to subscription",
 		"subscription_id", sub.ID,
 		"addon_id", req.AddonID,
+		"addon_type", a.Addon.Type,
 		"prices_count", len(validPrices),
-		"line_items_count", len(lineItems),
+		"line_items_created", len(lineItems),
+		"line_items_reused", len(existingActiveLineItems),
 	)
 
 	return addonAssociation, nil
 }
 
+// validateEntitlementCompatibility checks if addon entitlements are compatible with existing subscription entitlements
+// It ensures that entitlements with the same feature ID have the same usage reset period
+func (s *subscriptionService) validateEntitlementCompatibility(ctx context.Context, subscriptionID, addonID string) error {
+	// Get entitlements for the addon we're trying to add first
+	entitlementService := NewEntitlementService(s.ServiceParams)
+	addonEntitlements, err := entitlementService.GetAddonEntitlements(ctx, addonID)
+	if err != nil {
+		return err
+	}
+
+	// Early return if addon has no entitlements
+	if len(addonEntitlements.Items) == 0 {
+		return nil
+	}
+
+	// Filter addon entitlements to only those with feature IDs and usage reset periods
+	relevantAddonEntitlements := make([]*dto.EntitlementResponse, 0, len(addonEntitlements.Items))
+	for _, addonEnt := range addonEntitlements.Items {
+		if addonEnt.FeatureID != "" && addonEnt.UsageResetPeriod != "" {
```

Comment:
can this be simply feature type check? I think all we want to check here is metered type entitlement check